packages/gatsby/src/schema/schema.js

```diff
@@ -717,6 +731,40 @@ const addCustomResolveFunctions = async ({ schemaComposer, parentSpan }) => {
   })
 }
 
+const determineSearchableFields = ({ schemaComposer, typeComposer }) => {
+  typeComposer.getFieldNames().forEach(fieldName => {
+    const field = typeComposer.getField(fieldName)
+    const extensions = typeComposer.getFieldExtensions(fieldName)
+    if (field.resolve) {
+      if (extensions.dateformat) {
+        typeComposer.extendFieldExtensions(fieldName, {
+          searchable: SEARCHABLE_ENUM.SEARCHABLE,
+          sortable: SORTABLE_ENUM.SORTABLE,
+          needsResolve: extensions.proxyFrom ? true : false,
```

Comment:
hmm this is a very specific scenario

Reply:
It is. I'm open to better logic here. The idea is that our dateformat should always be searchable, even with args. Because our resolvers are ultimately functions, I'm not sure what would be better logic here.

Reply:
what happens when `dateformat` is combined with an extension other than `proxy` that also needs resolving, e.g. `projection` or similar? maybe these enums should apply per-resolver/extension, not per-field, since a field can have multiple wrapped resolvers?